src/utils/dataMapper.ts

```diff
@@ -55,20 +52,30 @@ export interface ResourceInput {
 }
 
 export interface Resource {
-  label: string,
+  labels: string[],
   name: string,
   description: string,
   uri: string,
   claimsGraphUri: string,
 }
 
 export function mapResources(selfDescriptions: ResourceInput): Resource[] {
-  console.log('From mapper: ', selfDescriptions);
-  return selfDescriptions.items.map(({ 'properties(n)': p, 'labels(n)': l }) => ({
-    label: l[1],
-    name: p.name,
-    description: p.description,
-    uri : p.uri,
-    claimsGraphUri : p.claimsGraphUri
-  }));
+  console.debug('mapResource: ', selfDescriptions);
+  return selfDescriptions
+    .items.map(({ 'properties(n)': p, 'labels(n)': l }) => ({
+      labels: l.filter(label => label !== 'Resource' && label !== 'DataResource'),
+      name: p.name,
+      description: p.description,
+      uri: p.uri,
+      claimsGraphUri: p.claimsGraphUri
+    }));
+}
+
+export interface ISelfDescription {
```

Comment:
We have interface definitions in ../types/.. We should have a standardised solution here. So move this interface to types/selfDescription.model?

Reply:
The `ISelfDescription` I consider to be an intermediate data type. We use it to interpret data received from the backend and to transform it into a new business related data type (at least we should do it in the feature). The difference between the two is that the `ISelfDescription` data type has a not **really usefull format**. We should converting it into a new type as all the other types were converted. 

For example take a look at the **src/utils/dataMapper.ts** file. you'll see `mapServiceOfferings` and `mapResources`. Both methods take in a data type received from a backend and convert them into a new type wich is more simple, more understandable, more easy to work with and describes much better a business concept and can be considered a model for that business concept. The returned types are the ones that belong to the model files inside the **types** folder.

I would not touch it in this pull request even though there are some problems around this type.  It is a nice catch though. You sensed also that is not quite well implementd. I think this would be a refactoring which in itself deserves a pull request. If you need a solution because of your current task, I can solve it for you in a new pull request or we can have a look together the you can solve it too.

Reply:
Thanks for the explanation. Then it can stay as it is for now